src/block/CaveVines.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\utils\SupportType;
+use pocketmine\data\runtime\RuntimeDataReader;
+use pocketmine\data\runtime\RuntimeDataWriter;
+use pocketmine\entity\Entity;
+use pocketmine\event\block\BlockGrowEvent;
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
+use pocketmine\item\VanillaItems;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\world\BlockTransaction;
+use pocketmine\world\sound\GlowBerriesPickSound;
+use function mt_rand;
+
+class CaveVines extends Flowable{
+	public const MAX_AGE = 25;
+
+	protected int $age = 0;
+
+	protected bool $hasBerries = false;
+
+	/** In Bedrock this called "head" */
+	protected bool $tip = false;
+
+	public function getRequiredStateDataBits() : int{
+		return 7;
+	}
+
+	public function describeState(RuntimeDataWriter|RuntimeDataReader $w) : void{
+		$w->boundedInt(5, 0, self::MAX_AGE, $this->age);
+		$w->bool($this->hasBerries);
+		$w->bool($this->tip);
+	}
+
+	public function readStateFromWorld() : Block{
+		parent::readStateFromWorld();
+
+		$this->tip = !$this->getSide(Facing::DOWN)->isSameType($this);
```

Comment:
This would be more appropriate for onNearbyBlockChange. Usually readStateFromWorld shouldn't be used unless working around some vanilla limitation.

Reply:
I don't use onNearbyBlockChange() here because it's not call when the block itself is placed or changed

Reply:
That's not correct... and even if you need placement logic, you have place() for that

Reply:
place() is not called when a block has been placed using world->setBlock()

Reply:
onNearbyBlockChange is, unless you disabled updating.

Reply:
onNearbyBlockChange is called in both of the cases you mentioned.